Document missing sysvars (#6485)
pingcap/docs: system-variables.md

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4`
- The default character set for the server.

### collation_connection

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4_bin`

> - The collation for string literals that do not have a specified collation.

Reviewer: ```suggestion
- This variable indicates the collation for string literals that do not have a specified collation.
```

### collation_database

- Scope: SESSION | GLOBAL
- Default value: `utf8mb4_bin`

> - This variable indicates the collation of the default database that is in use. **It is NOT recommended to set this variable**. When a new default database is selected, the server changes the variable value.

Reviewer: ```suggestion
- This variable indicates the collation of the default database in use. **It is NOT recommended to set this variable**. When a new default database is selected, the server changes the variable value.
```

- This variable sets the authentication method that the server advertises when the server-client connection is being established. Possible values for this variable are documented in [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).
- Value options: `mysql_native_password` and `caching_sha2_password`. For more details, see [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).

### default_week_format

- Scope: SESSION | GLOBAL
- Default value: `0`
- Range: `[0, 7]`

> - Set the week format used by the `WEEK()` function

Reviewer: ```suggestion
- Sets the week format used by the `WEEK()` function.
```

### foreign_key_checks

- Scope: SESSION | GLOBAL
- Default value: `OFF`
- For compatibility, TiDB returns foreign key checks as `OFF`.

### group_concat_max_len

- Scope: SESSION | GLOBAL
- Default value: `1024`
- Range: `[4, 18446744073709551615]`
- The maximum buffer size for items in the `GROUP_CONCAT()` function.

### have_openssl

- Scope: NONE
- Default value: `DISABLED`

> - MySQL compatibiltiy read-only variable. Set to `YES` when the server has TLS enabled.

Reviewer: ```suggestion
- A read-only variable for MySQL compatibility. You can set it to `YES` when the server has TLS enabled.
```

Author: This is set by the server, not by the user.

### have_ssl

- Scope: NONE
- Default value: `DISABLED`

> - MySQL compatibiltiy read-only variable. Set to `YES` when the server has TLS enabled.

Reviewer: ```suggestion
- A read-only variable for MySQL compatibility. You can set it to `YES` when the server has TLS enabled.
```

Author: This is set by the server, not by the user.

- This variable sets the authentication method that the server advertises when the server-client connection is being established. Possible values for this variable are documented in [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).
- Value options: `mysql_native_password` and `caching_sha2_password`. For more details, see [Authentication plugin status](/security-compatibility-with-mysql.md#authentication-plugin-status).

### default_week-format

- Scope: SESSION | GLOBAL
- Default value: `0`
- Range: `[0, 7]`
- Set the week format used by the `WEEK()` function

### error_count

- Scope: SESSION
- Default value: `0`

> - This variable exists only for MySQL compatibility and is always `0` in TiDB.

Reviewer 2: It looks like there is code in the server for this feature, but it might not be working correctly?

Author: ```
mysql-8.0.26 > select now(999999);
ERROR 1426 (42000): Too-big precision 63 specified for 'now'. Maximum is 6.
mysql-8.0.26 > SELECT @@session.error_count;
+-----------------------+
| @@session.error_count |
+-----------------------+
|                     1 |
+-----------------------+
1 row in set (0.00 sec)
```

```
5.7.25-TiDB-v5.0.4 127.0.0.1:4000  test  SQL  select now(999999);
ERROR: 1105 (HY000): Too-big precision 999999 specified for 'now'. Maximum is 6.
5.7.25-TiDB-v5.0.4 127.0.0.1:4000  test  SQL  SELECT @@session.error_count;
+-----------------------+
| @@session.error_count |
+-----------------------+
| 0                     |
+-----------------------+
1 row in set (0.0014 sec)
```

```
8.0.26 127.0.0.1:8026   test  SQL  select now(999999);
ERROR: 1426 (42000): Too-big precision 63 specified for 'now'. Maximum is 6.
8.0.26 127.0.0.1:8026   test  SQL  SELECT @@session.error_count;
+-----------------------+
| @@session.error_count |
+-----------------------+
|                     0 |
+-----------------------+
1 row in set (0.0007 sec)
```

```
mysql> select now(999999);
ERROR 1105 (HY000): Too-big precision 999999 specified for 'now'. Maximum is 6.
mysql> SELECT @@session.error_count;
+-----------------------+
| @@session.error_count |
+-----------------------+
| 0                     |
+-----------------------+
1 row in set (0.00 sec)
```

Looks like this doesn't work with MySQL Shell, but also not with TiDB.

### foreign_key_checks

- Scope: SESSION | GLOBAL
- Default value: `OFF`
- For compatibility, TiDB returns foreign key checks as `OFF`.

### group_concat_max_len

- Scope: SESSION | GLOBAL
- Default value: `1024`
- Range: `[4, 18446744073709551615]`

> - The maximum number of elements for the `GROUP_CONCAT()` function.

Reviewer 2: It's buffer not elements, no?
```suggestion
- The maximum buffer size for items in the `GROUP_CONCAT()` function.
```

- Unit: Seconds
- This variable represents the idle timeout of the interactive user session, which is measured in seconds. Interactive user session refers to the session established by calling [`mysql_real_connect()`](https://dev.mysql.com/doc/c-api/5.7/en/mysql-real-connect.html) API using the `CLIENT_INTERACTIVE` option (for example, MySQL shell client). This variable is fully compatible with MySQL.

### last_insert_id

- Scope: SESSION
- Default value: ""

> - MySQL Compatibility variable. Not used in TiDB. Use the `LAST_INSERT_ID()` function instead.

Reviewer 2: I think we should fix this (it is easy to do with a SetSession func) rather than document it as broken.

Some connectors may require it.

Author: Could you open an issue for this? Then we can update the docs once it is fixed.

Reviewer 2: https://github.com/pingcap/tidb/pull/28072

- Default value: `Apache License 2.0`
- This variable indicates the license of your TiDB server installation.

### max_allowed_packet

- Scope: GLOBAL | SESSION
- Default value: `67108864`
- Range: `[1024, 1073741824]`

> - Maximum size of a packet for the MySQL protocol

Reviewer: ```suggestion
- The maximum size of a packet for the MySQL protocol.
```

- Default value: `Apache License 2.0`
- This variable indicates the license of your TiDB server installation.

### max_allowed_packet

- Scope: GLOBAL | SESSION
- Default value: `67108864`
- Range: `[1024, 1073741824]`
- Maximum size of a packet for the MySQL protocol

### max_connections

- Scope: GLOBAL
- Default value: `151
- Range: `[1, 100000]`

> - MySQL Compatibility variable. Not used in TiDB.

Reviewer 2: It looks like this should be a noop sysvar (there is a concept of max connections in tidb; but it uses the config file to set).

So far we've not been documenting noop sysvars. I think until we've documented all the actual sysvars we should stick to this strategy, otherwise we'll get tied down in reviews for documentation that is not useful.

- Range: `[0, 18446744073709551615]`
- The maximum number of rows returned by the `SELECT` statements.

### ssl_ca

- Scope: NONE
- Default value: ""

> - The location of the certificate authority file if specified on startup.

Reviewer: ```suggestion
- The location of the certificate authority file (if there is one) when TiDB is started.
```

### ssl_cert

- Scope: NONE
- Default value: ""

> - The location of the certificate file that is used for SSL/TLS connections if specified on startup.

Reviewer: ```suggestion
- The location of the certificate file (if there is a file) that is used for SSL/TLS connections when TiDB is started.
```

### ssl_key

- Scope: NONE
- Default value: ""

> - The location of the private key file that is used for SSL/TLS connections if specified on startup.

Reviewer: ```suggestion
- The location of the private key file (if there is one) that is used for SSL/TLS connections when TiDB is started.
```
